### PR TITLE
Update XNNPACK to commit b0e2ac129b0952dd070889fade85aa3943308a61

### PR DESCRIPTION
**Pull request by mcr229**
Updating XNNPACK Commit to b0e2ac129b0952dd070889fade85aa3943308a61.

This PR contains the following changes:
- xnnpack commit --> b0e2ac129b0952dd070889fade85aa3943308a61
- cpuinfo commit --> d6860c477c99f1fce9e28eb206891af3c0e1a1d7
- pthreadpool commit -->4fe0e1e183925bf8cfa6aae24237e724a96479b

Additionally we add new microkernels:
- NEONI8MM_MICROKERNELS
- ARVXVNNI_MICROKERNELS
- AVX512VBMI_MICROKERNELS
- Regenerate microkernel wrappers
- Create microkernel_config buck target library
    - This is because microkernel_configs were all refactored into src/configs/...

### CODE REVIEW

File: backends/xnnpack/third-party/xnnpack.buck.bzl [hunk 1]
```diff
@@ -980,6 +1038,48 @@ def define_xnnpack():
...
+            "ovr_config//cpu:x86_64": DEFAULT_DUMMY_SRC,
+        }),
+        headers = subdir_glob([
+            ("XNNPACK/src", "**/*.h"),
+            ("XNNPACK/src", "**/*.c"),
+        ]),
+        header_namespace = "",
+        compiler_flags = [
+            "-O2",
+            "-Wno-error=missing-braces",  # required since the SGX toolchain does not have this by default
```
Comment (digantdesai) [inline backends/xnnpack/third-party/xnnpack.buck.bzl:1066]: This is why we need to just copy from bazel :P

File: backends/xnnpack/third-party/xnnpack.buck.bzl [hunk 2]
```diff
@@ -1055,9 +1155,13 @@ def define_xnnpack():
             "-DXNN_ENABLE_ASSEMBLY",
             "-DXNN_ENABLE_GEMM_M_SPECIALIZATION",
             "-DXNN_ENABLE_ARM_DOTPROD",
+            "-DXNN_ENABLE_CPUINFO",
+            "-DXNN_ENABLE_DWCONV_MULTIPLASS=1",
```
Comment (digantdesai) [inline backends/xnnpack/third-party/xnnpack.buck.bzl:1159]: do we use this?

File: backends/xnnpack/third-party/xnnpack_src_defs.bzl
Comment (digantdesai) [inline backends/xnnpack/third-party/xnnpack_src_defs.bzl:1]: can't even open this file

**Review by digantdesai — APPROVED** [review]

File: backends/xnnpack/third-party/generate-xnnpack-wrappers.py [hunk 3]
```diff
@@ -8,16 +8,19 @@
 
 BANNER = "Auto-generated by generate-wrappers.py script. Do not modify"
 WRAPPER_SRC_NAMES = {
```
Comment (mcr229) [inline backends/xnnpack/third-party/generate-xnnpack-wrappers.py:10]: @digantdesai I retook inventory of all the prod microkernel sources and sources that we were/weren't using and cleaned this up. Also added new ones that the new update introduces. Do you mind taking a look here

File: backends/xnnpack/third-party/xnnpack.buck.bzl [hunk 4]
```diff
@@ -1,38 +1,47 @@
 load("//third-party:glob_defs.bzl", "subdir_glob")
 load(
     ":xnnpack_src_defs.bzl",
-    "HOT_SRCS",
```
Comment (mcr229) [inline backends/xnnpack/third-party/xnnpack.buck.bzl:4]: @digantdesai I also did a lot of clean up on this buck.bzl file, do you mind taking a look at this one as well.

File: backends/xnnpack/third-party/xnnpack.buck.bzl (hunk 2 above)
Comment (mcr229) [reply backends/xnnpack/third-party/xnnpack.buck.bzl:1198]: https://github.com/google/XNNPACK/blob/7967e2c8a6967bbe8dd4c627d07062372fadcf60/src/configs/gemm-config.c#L94

kinda dumb, but this gemm-config requires cpuinfo, but for some reason disabling cpuinfo means we will not include it into the file.

Comment (mcr229) [reply backends/xnnpack/third-party/xnnpack.buck.bzl:1198]: However, we technically can turn off the dwconv_multipass

File: backends/xnnpack/third-party/xnnpack.buck.bzl [hunk 5]
```diff
@@ -101,12 +102,68 @@ def define_xnnpack():
...
+            "-Wno-error=missing-braces",  # required since the SGX toolchain does not have this by default
+        ],
+        preferred_linkage = "static",
+        preprocessor_flags = [
+            "-DXNN_ENABLE_CPUINFO=1",
+            # "-DXNN_ENABLE_JIT=0",
+            "-DXNN_ENABLE_DWCONV_MULTIPLASS=1",
+            "-DXNN_ENABLE_GEMM_M_SPECIALIZATION=1",
+            "-DXNN_ENABLE_ARM_I8MM=1",
+            "-DXNN_ENABLE_ARM_DOTPROD",
```
Comment (digantdesai) [inline backends/xnnpack/third-party/xnnpack.buck.bzl]: =1?

**Pull request closed without merging**